extra_geom/base.py

```diff
@@ -54,6 +55,12 @@ def offset(self, shift):
         pos = self.corner_pos + shift
         return type(self)(pos, self.ss_vec, self.fs_vec, self.ss_pixels, self.fs_pixels)
 
+    def rotate(self, angles, center):
```

Comment:
The `angles` parameter here should probably be called something like `matrix`, right?